ui/src/app-bar/app-bar.tsx

```diff
@@ -6,18 +6,29 @@ import vaylaLogo from 'vayla-design-lib/logo/vayla-logo.svg';
 import { EnvRestricted } from 'environment/env-restricted';
 import { Environment } from 'environment/environment-info';
 import { useTranslation } from 'react-i18next';
+import { useInfraModelAppSelector } from 'store/hooks';
+import { GeometryPlanId } from 'geometry/geometry-model';
+import { inframodelEditPath } from 'infra-model/infra-model-main-view';
 
 type Link = {
     link: string;
     name: string;
     type: Environment;
 };
 
-const links: Link[] = [
+const links: (a: GeometryPlanId | undefined) => Link[] = (
+    inframodelBeingEdited: GeometryPlanId | undefined,
+) => [
     { link: '/', name: 'app-bar.frontpage', type: 'prod' },
     { link: '/track-layout', name: 'app-bar.track-layout', type: 'prod' },
     { link: '/registry', name: 'app-bar.register', type: 'test' },
-    { link: '/infra-model', name: 'app-bar.infra-model', type: 'prod' },
+    {
+        link: `/infra-model${
+            inframodelBeingEdited ? `${inframodelEditPath}/${inframodelBeingEdited}` : ''
+        }`,
+        name: 'app-bar.infra-model',
+        type: 'prod',
+    },
```

Comment:
🥩 . Valittu inframalli pidetään aina tallessa reduxissa, joten this was the easy part